packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -799,9 +799,14 @@ class _DraggableScrollableSheetScrollPosition extends ScrollPositionWithSingleCo
   _DraggableScrollableSheetScrollPosition({
     required super.physics,
     required super.context,
-    super.oldPosition,
+    ScrollPosition? oldPosition,
     required this.getExtent,
-  });
+  }) : super(oldPosition: oldPosition) {
+    if (oldPosition is _DraggableScrollableSheetScrollPosition
```

Comment:
Do we know that this will always be true? Since this is private implementation?

Reply:
Actually, this is not always true.
This may be another type of position when the widget type changes.
Make a type check here also cause some tests failed.

Reply:
Hmm, is there another way we could indicate this? Perhaps a flag?
Using `is` here is not congruent with the style guide: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-interleaving-multiple-concepts-together

Reply:
oops, this is a bit stumbling for me.
I took a look and here is the only chance to catch old position information.
I agree with the above guidance, but here is just to distinguish whether the old type is consistent with myself, and does not interleave other concepts, I think it is reasonable, what do you think? ： ）

Reply:
I dunno, tbh I am not very familiar with DraggableScrollableSheet, @dnfield may know more.
All of this is private implementation, so it might be ok to distinguish this case.
What tests failed if the type is assumed to be _DraggableScrollableSheetScrollPosition?

I'll reach out to @Hixie on Discord in case this is a style guide exception.

Reply:
I checked to see what tests failed - mostly  `'ScrollPositionWithSingleContext' is not a subtype of type
'_DraggableScrollableSheetScrollPosition?' in type cast`

(I was adding a type cast in the createPosition function of the controller)

What about adding a copyWith to _DraggableScrollableSheetScrollPosition, that way when it is created, you can copy over any ScrollPosition related info (including subclasses like ScrollPositionWithSingleContext)? Just throwing out ideas.. IIRC the copyWith in ScrollBehavior was due to a similar situation. I saw you landed a fix in that recently by the way - thanks!

Reply:
@Piinks Thank you so much for catching this bad smell 😄 
 I think with your ideas I found the perfect solution.